python_modules/dagster/dagster/core/asset_defs/materialize.py

```diff
@@ -40,3 +42,43 @@ def materialize(
         assets=assets_defs,
         source_assets=source_assets,
     ).execute_in_process(run_config=run_config, instance=instance)
+
+
+def materialize_in_process(
+    assets: Sequence[Union[AssetsDefinition, SourceAsset]],
+    run_config: Any = None,
+    instance: Optional[DagsterInstance] = None,
+    resources: Optional[Mapping[str, object]] = None,
+) -> ExecuteInProcessResult:
+    """
+    Executes a single-threaded, in-process run which materializes provided assets.
+
+    By default, will materialize assets to memory, meaning results will not be
+    persisted. This behavior can be changed by overriding the default io
+    manager key "io_manager", or providing custom io manager keys to assets.
```

Comment:
If someone provides an io_manager_key but doesn't provide any resources for it, will this throw raise an error? I think ideally we'd use the in-memory IO manager.